### ISSUE TITLE
CRD managed pods not shown in the UI in Group by Namespace/Workload view

### ISSUE THREAD

**Issue by dnoland1** [rancher/rancher#22106]
**What kind of request is this (question/bug/enhancement/feature request):**
Enhancement

**Steps to reproduce (least amount of steps as possible):**
Launch pods managed by a CRD and go to the Workloads page. Can use this example from Elasticsearch (https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-quickstart.html) by running:
```
$ kubectl apply -f https://download.elastic.co/downloads/eck/0.9.0/all-in-one.yaml
$ cat <<EOF | kubectl apply -f -
apiVersion: elasticsearch.k8s.elastic.co/v1alpha1
kind: Elasticsearch
metadata:
  name: quickstart
spec:
  version: 7.2.0
  nodes:
  - nodeCount: 1
    config:
      node.master: true
      node.data: true
      node.ingest: true
EOF
```
**Result:**
It appears the UI or API will filter out all pods that are not managed by ReplicaSets, DaemonSets, StatefulSets, or unmanaged. If a pod is managed by a CRD, it is not displayed in the UI. The above example will set the following in the pod:
```yaml
  ownerReferences:
  - apiVersion: elasticsearch.k8s.elastic.co/v1alpha1
    blockOwnerDeletion: true
    controller: true
    kind: Elasticsearch
    name: quickstart
    uid: 59253a84-ba36-11e9-a46f-0af107b47750
```

**Other details that may be helpful:**
After running above YAML and also launching an unmanaged pod, see this in CLI:
```
$ kubectl get pods
NAME                       READY   STATUS    RESTARTS   AGE
quickstart-es-cd28s99bh4   1/1     Running   0          22m
ubuntu-pod                 1/1     Running   0          17m
```
But see this in UI:
![image](https://user-images.githubusercontent.com/468462/62813077-f05e7280-bad6-11e9-9dc8-17983b7cf869.png)

For showing these pods in the UI, would be good to note them as CRD managed and possibly give information on which CRD is managing the pod.

This issue may be somewhat related - https://github.com/rancher/rancher/issues/13323
**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
v2.2.7
- Installation option (single install/HA):
N/A

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported):
Custom
- Machine type (cloud/VM/metal) and specifications (CPU/memory):
AWS, t3a.medium
- Kubernetes version (use `kubectl version`):
v1.14.5

- Docker version (use `docker version`):
N/A

**Comment by superseb** [comment]
Previously fixed in https://github.com/rancher/rancher/issues/15804

**Comment by fblgit** [comment]
Actually, this stills happening.
Deploy etcd-operator
Build a example cluster.
Pods not shown in UI.

How to reproduce:

kubectl create -f https://coreos.com/operators/etcd/latest/deployment.yaml
kubectl create -f https://coreos.com/operators/etcd/latest/example-etcd-cluster.yaml

**Comment by dnoland1** [comment]
@superseb So this appears to be a regression/bug and not a feature request, correct?

**Comment by SkySonR** [comment]
The same think with official jenkins-operator CRD:
https://github.com/jenkinsci/kubernetes-operator

**Comment by prachidamle** [comment]
@SkySonR I tested this on latest rancher, but I can see "jenkins-operator" listed under workloads.
I followed the intstallation instructions here:
https://github.com/jenkinsci/kubernetes-operator/blob/master/documentation/installation.md

<img width="1069" alt="Screen Shot 2019-10-04 at 1 52 50 PM" src="https://user-images.githubusercontent.com/1470602/66239301-6255d100-e6ae-11e9-9d8b-118fe4a7e569.png">

**Comment by dnoland1** [comment]
@prachidamle That operator is a k8s deployment and shows up fine inthe UI. Believe it will watch for the creation a jenkins crd. Question is, if you create a jenkins crd and the jenkins-operator creates pods, do those pods show in the UI?

Or try out my elasticsearch example in the original post as a test case.

**Comment by prachidamle** [comment]
@dnoland1 I tried out the elasticsearch example in the original post. Here is what I see:

1) I see the "quickstart-es-xyz" pod in the UI in the flat list Or in the "Group by node" view.
2) I guess since this is not a pod that belongs to a "workload" in Rancher (having type Deployments, DaemonSets, StatefulSets, or CronJobs), it is not listed on "Group By Workload" view. But it is listed in the pods list.

<img width="1316" alt="Screen Shot 2019-10-04 at 2 45 50 PM" src="https://user-images.githubusercontent.com/1470602/66242130-c62fc800-e6b5-11e9-8f92-e64124ef7f05.png">

<img width="1311" alt="Screen Shot 2019-10-04 at 2 45 23 PM" src="https://user-images.githubusercontent.com/1470602/66242138-cd56d600-e6b5-11e9-9777-7ce9ad1feb55.png">

**Comment by dnoland1** [comment]
@prachidamle What if you just deploy a simple ubuntu pod using YAML as I did above (see ubuntu-pod in screen capture)? That ubuntu-pod does not belong to a deployment, daemonset, statefulset, or cronjob, but is still displayed in the Group by Namespace/Workload view. Can try this YAML:
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: ubuntu-pod
spec:
  containers:
  - name: "ubuntu-pod"
    image: "ubuntu:latest"
    command:
      - /bin/sleep
      - "10000"
```

**Comment by prachidamle** [comment]
@dnoland1 I tried that out, so looks like UI is listing the unmanaged pod on the Group by Namespace/Workload view - even if API is not actually listing that unmanaged pod under workloads.

If you see the /workloads API list then the ubuntu unmanaged pod is not present in that output. It is however listed under /pods. 

So looks like UI is listing that pod under workloads - if we also want to show the CRD backed pods, we probably need a similar shift on the UI display applied.

<img width="912" alt="Screen Shot 2019-10-04 at 3 23 07 PM" src="https://user-images.githubusercontent.com/1470602/66243953-44429d80-e6bb-11e9-99df-f30ba4797bfc.png">
<img width="1285" alt="Screen Shot 2019-10-04 at 3 21 14 PM" src="https://user-images.githubusercontent.com/1470602/66243954-44429d80-e6bb-11e9-8993-1630d7fe8da5.png">
<img width="1294" alt="Screen Shot 2019-10-04 at 3 21 04 PM" src="https://user-images.githubusercontent.com/1470602/66243955-44429d80-e6bb-11e9-990e-3229827058c0.png">

**Comment by prachidamle** [comment]
@dnoland1 Looks like we set `workloadId` on the Pod even if there is no workload owning it - that causes UI to not show in the "group by workload" view since associated workload cannot be found.

The fix here is not setting `workloadId` on the Pod managed by a CRD so that it shows up just like the standalone unmanaged pod.

**Comment by averri** [comment]
I see the same issue happening. I have tested with https://www.arangodb.com/docs/3.5/tutorials-kubernetes.html

The pods in Kubernetes dashboard:

![image](https://user-images.githubusercontent.com/1898447/69834144-80642b80-1230-11ea-9984-1351b57b311c.png)

... they does not appear on Rancher UI:

![image (1)](https://user-images.githubusercontent.com/1898447/69834152-8a862a00-1230-11ea-835a-d4eafd5a3f6d.png)

**Comment by phal0r** [comment]
I can confirm @averri 's findings. Using ArangoDB Operator to deploy an ArangoCluster works fine, with the exception, that these pods are not shown in the ui.

**Comment by jiaqiluo** [comment]
The bug is reproduced in Rancher:v2.3.3

The steps are the same as the issue provides

Results:
- the pods show in the UI in the `flat list` or in the "Group by node" view, but not in the `group by workload` or `group by namespace/workload` view
<img width="1461" alt="Screen Shot 2019-12-11 at 2 26 02 PM" src="https://user-images.githubusercontent.com/6218999/70661752-344abb00-1c22-11ea-950b-fd7c4c3a0300.png">

**Comment by jiaqiluo** [comment]
the bug fix is validated in master-head 509b961a4

The steps are the same as the issue provides

Results:
- now we can see the pod under all four views
<img width="1314" alt="Screen Shot 2019-12-11 at 2 38 53 PM" src="https://user-images.githubusercontent.com/6218999/70662918-796fec80-1c24-11ea-9885-6dc01aa6a156.png">
<img width="1321" alt="Screen Shot 2019-12-11 at 2 38 57 PM" src="https://user-images.githubusercontent.com/6218999/70662919-796fec80-1c24-11ea-906d-93537109f7a9.png">
<img width="1311" alt="Screen Shot 2019-12-11 at 2 44 09 PM" src="https://user-images.githubusercontent.com/6218999/70663036-b89e3d80-1c24-11ea-9ce3-13995127d1ab.png">
<img width="1316" alt="Screen Shot 2019-12-11 at 2 44 14 PM" src="https://user-images.githubusercontent.com/6218999/70663037-b89e3d80-1c24-11ea-8d1a-342ded8d2a09.png">


- `workloadId` is set to NULL
<img width="449" alt="Screen Shot 2019-12-11 at 2 39 13 PM" src="https://user-images.githubusercontent.com/6218999/70662955-8ab8f900-1c24-11ea-8477-4e81b013718b.png">
